src/coreclr/jit/emit.cpp

```diff
@@ -3071,8 +3071,21 @@ void emitter::emitSplit(emitLocation*         startLoc,
 
     } // end for loop
 
-    splitIfNecessary();
-    assert(curSize < UW_MAX_FRAGMENT_SIZE_BYTES);
+    if ((igLastCandidate != nullptr) && (curSize == candidateSize))
+    {
+        // TODO: This scenario may be possible (but very unlikely) when using the default fragment size.
+        // Fixing this would likely require padding out the last fragment to not be zero-sized, and doing the split.
+        // For now, assume this only occurs under stress modes, or when using DOTNET_JitSplitFunctionSize.
+        assert(maxSplitSize != UW_MAX_FRAGMENT_SIZE_BYTES);
```

Comment:
I don't think this comment is quite right. Yes, it is unlikely. But we're talking about a single empty fragment at the end. If there was a non-empty fragment, then `curSize != candidateSize`. So it seems like the fix works for any split size, including the default, and doesn't require this assert or `TODO`. (It might require other commenting/documentation, but that's a separate question.)

Reply:
> But we're talking about a single empty fragment at the end.

Yeah, on second thought, I think you're right. If we have a 512KB-sized IG, followed by an empty IG that happens to be zero-sized, then the last fragment should be exactly 512KB in size, which shouldn't break any unwind data invariants. I'll remove the TODO.